client/src/components/ui/contextmenu/ContextTemplate.svelte

```diff
@@ -1,27 +1,52 @@
 <script lang="ts">
+    import { Events } from '../../types.ts';
     export let show = false;
+    
+    let context: HTMLDialogElement | null;
+
+    $: if (context && show) context?.showModal();
+
+    function onClose() {
+        show = false;
+        dispatch(Events.ContextClose)
+    }
 </script>
 
-<div class="drawer" class:show={show}>
-    <slot></slot>
-</div>
+<dialog
+    class:show = {show}
+    bind:this = {context}
+    on:close = {() => onClose()}
+    on:click|self = {() => context?.close()}
+    on:keydown|self = {() => context?.close()}
+    >
+    <div class="drawer"
+        on:click|stopPropagation
+        on:keydown|stopPropagation
+    >
+        <slot></slot>
+    </div>
+</dialog>
 
 <style>
-    @import url('../../../pages/vars.css');
+    @import url('../../../pages/global.css');
 
-    div {
+    dialog {
         width: 100vw;
         border: var(--spacing-tiny) solid black;
-        transition: bottom var(--animation-length);
-        bottom: -100%;
+        bottom: -95%;
```

Comment:
Need help.

```suggestion
        bottom: -95%;
```
As of now this is hardcoded, setting this to 100% for `dialog` tags seems to affix it to the pixel center of the dialog tag unlike `div` which affixes the bottom most part to the bottom of the page.

This will break if there are >2 selections in the context menu pretty much,

Reply:
The CSS refactor can be addressed in a future. Marking this as resolved for now.